modules/vba_documents/README.yml

```diff
@@ -0,0 +1,342 @@
+info:
+  version: 0.0.0
+  title: Benefits
+  description: |
+    Veterans Benefits Administration (VBA) document uploads.
+    
+    ## Background
+    
+    This API is being provided as a proof of concept of a general-purpose VA 
+    API gateway. The use case is to allow authorized third-party developers to 
+    submit VBA documents and have a high degree of assurance that those documents 
+    will be processed and stored in the appropriate VA data system.
+    
+    The Document Upload API passes data through to ICMHS aka “Central Mail API”.
+    Central Mail API accepts a payload consisting of a document in PDF format, zero or more
+    optional attachments in PDF format, and some JSON metadata. The metadata describes the
+    document, attachments, and identifies the person that the document relates to. This payload is
+    encoded as multipart/form-data. A unique identifier supplied with the payload can be used to
+    subsequently request the processing status of the uploaded document package.
+    
+    ## Design
+    
+    ### Authorization
+    
+    API requests are authorized by means of a symmetric API token, provided in an HTTP header
+    with name "api_key".
+
+    ### Upload Operation
+    
+    Allows a client to upload a document package (form + attachments + metadata).
+
+    1. Client Request: POST https://api.vets.gov/services/vba_documents/v0/
+        * No request body or parameters required
+    
+    2. Service Response: A JSON API object with the following attributes:
+        * `guid`: An identifier that can be used for subsequent status requests
+        * `location`: A URL to which the actual document package payload can be submitted
+          in the next step. The URL is specific to this upload request and should not be
+          re-used for subsequent uploads. The URL is valid for 900 seconds (15 minutes)
+          from the time of this response.
+          
+    3. Client Request: PUT to the location URL returned in Step 2.
+        * Request body should be encoded as multipart/form-data, equivalent to that
+          generated by an HTML form submission or using “curl -F…”. The format is
+          described in more detail below.
+        * No `api_key` authorization header is required for this request, as authorization is
+          embedded in the signed location URL.
+          
+    4. Service Response: The HTTP status indicates whether the upload was successful.
+      Additionally the response includes an ETag header containing an MD5 hash of the
+      submitted payload. This can be compared to the submitted payload to ensure data
+      integrity of the upload
+
+      ## Reference
+
+      Raw Open API Spec: http://dev-api.vets.gov/services/vba_documents/docs/v0/api
+
+  termsOfService: ''
+  contact:
+    name: Vets.gov
+tags:
+  - name: document_uploads
+    description: VA Benefits document upload functionality
+servers: 
+  - url: dev-api.vets.gov/services/vba_benefits/{version}
```

Comment:
May want to inject  `Settings.hostname` here somehow so it's `dev-api/staging/api/api` as appropriate. Maybe could be a responsibility of the controller.

Reply:
May also need to monkey with the `scheme` parameter to make it work via http/https